### ISSUE TITLE
DISABLED test_torchscript_function_exception (__main__.JitRpcTestWithSpawn)

### ISSUE THREAD

**Issue by mrshenli** [pytorch/pytorch#34353]
https://app.circleci.com/jobs/github/pytorch/pytorch/4716746

```
Mar 06 03:33:19 ...[E request_callback.cpp:39] Received error while processing request type 0: attempted to get undefined function python_function (get_function at /var/lib/jenkins/workspace/torch/csrc/jit/api/compilation_unit.h:69)
Mar 06 03:33:19 frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0x6a (0x7fd00962479a in /opt/conda/lib/python3.6/site-packages/torch/lib/libc10.so)
Mar 06 03:33:19 frame #1: <unknown function> + 0x65bb3d (0x7fd00a220b3d in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 06 03:33:19 frame #2: torch::distributed::rpc::RequestCallbackImpl::processRpc(torch::distributed::rpc::RpcCommandBase&, torch::distributed::rpc::MessageType, long) const + 0x18dc (0x7fd00a467b6c in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 06 03:33:19 frame #3: torch::distributed::rpc::RequestCallbackImpl::processMessage(torch::distributed::rpc::Message&) const + 0x55 (0x7fd00a468dd5 in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 06 03:33:19 frame #4: torch::distributed::rpc::RequestCallback::operator()(torch::distributed::rpc::Message&) const + 0x33 (0x7fd006f34583 in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_cpu.so)
Mar 06 03:33:19 frame #5: <unknown function> + 0x884b9b (0x7fd00a449b9b in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 06 03:33:19 frame #6: c10::ThreadPool::main_loop(unsigned long) + 0x2fb (0x7fd009615cab in /opt/conda/lib/python3.6/site-packages/torch/lib/libc10.so)
Mar 06 03:33:19 frame #7: <unknown function> + 0xc8421 (0x7fd009b19421 in /opt/conda/lib/libstdc++.so.6)
Mar 06 03:33:19 frame #8: <unknown function> + 0x76ba (0x7fd0127eb6ba in /lib/x86_64-linux-gnu/libpthread.so.0)
Mar 06 03:33:19 frame #9: clone + 0x6d (0x7fd01252141d in /lib/x86_64-linux-gnu/libc.so.6)
Mar 06 03:33:19 
Mar 06 03:33:20 .[E request_callback.cpp:39] Received error while processing request type 0: attempted to get undefined function nonexisting_script (get_function at /var/lib/jenkins/workspace/torch/csrc/jit/api/compilation_unit.h:69)
Mar 06 03:33:20 frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0x6a (0x7f919d1f579a in /opt/conda/lib/python3.6/site-packages/torch/lib/libc10.so)
Mar 06 03:33:20 frame #1: <unknown function> + 0x65bb3d (0x7f919ddf1b3d in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 06 03:33:20 frame #2: torch::distributed::rpc::RequestCallbackImpl::processRpc(torch::distributed::rpc::RpcCommandBase&, torch::distributed::rpc::MessageType, long) const + 0x18dc (0x7f919e038b6c in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 06 03:33:20 frame #3: torch::distributed::rpc::RequestCallbackImpl::processMessage(torch::distributed::rpc::Message&) const + 0x55 (0x7f919e039dd5 in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 06 03:33:20 frame #4: torch::distributed::rpc::RequestCallback::operator()(torch::distributed::rpc::Message&) const + 0x33 (0x7f919ab05583 in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_cpu.so)
Mar 06 03:33:20 frame #5: <unknown function> + 0x884b9b (0x7f919e01ab9b in /opt/conda/lib/python3.6/site-packages/torch/lib/libtorch_python.so)
Mar 06 03:33:20 frame #6: c10::ThreadPool::main_loop(unsigned long) + 0x2fb (0x7f919d1e6cab in /opt/conda/lib/python3.6/site-packages/torch/lib/libc10.so)
Mar 06 03:33:20 frame #7: <unknown function> + 0xc8421 (0x7f919d6ea421 in /opt/conda/lib/libstdc++.so.6)
Mar 06 03:33:20 frame #8: <unknown function> + 0x76ba (0x7f91a63bc6ba in /lib/x86_64-linux-gnu/libpthread.so.0)
Mar 06 03:33:20 frame #9: clone + 0x6d (0x7f91a60f241d in /lib/x86_64-linux-gnu/libc.so.6)
Mar 06 03:33:20 
Mar 06 03:33:22 .[E request_callback.cpp:39] Received error while processing request type 0: The following operation failed in the TorchScript interpreter.
Mar 06 03:33:22 Traceback of TorchScript (most recent call last):
Mar 06 03:33:22   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/jit/rpc_test.py", line 140, in raise_script
Mar 06 03:33:22 @torch.jit.script
Mar 06 03:33:22 def raise_script():
Mar 06 03:33:22     raise RuntimeError("Expected error")
Mar 06 03:33:22     ~~~~~~~~~~~~~~~~~~~~ <--- HERE
Mar 06 03:33:22     return 0
Mar 06 03:33:22 RuntimeError: Exception
Mar 06 03:33:22 
Mar 06 03:34:36 .............sProcess process 3:
Mar 06 03:34:36 Traceback (most recent call last):
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
Mar 06 03:34:36     self.run()
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/multiprocessing/process.py", line 93, in run
Mar 06 03:34:36     self._target(*self._args, **self._kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 243, in _run
Mar 06 03:34:36     getattr(self, test_name)()
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/dist_utils.py", line 71, in new_test_method
Mar 06 03:34:36     rpc.shutdown(graceful=clean_shutdown)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 76, in wrapper
Mar 06 03:34:36     return func(*args, **kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 237, in shutdown
Mar 06 03:34:36     _wait_all_workers()
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 76, in wrapper
Mar 06 03:34:36     return func(*args, **kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 164, in _wait_all_workers
Mar 06 03:34:36     args=(sequence_id, self_worker_name,),
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 76, in wrapper
Mar 06 03:34:36     return func(*args, **kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 526, in rpc_sync
Mar 06 03:34:36     return fut.wait()
Mar 06 03:34:36 RuntimeError: RPC ran for more than 60000 milliseconds and timed out.
Mar 06 03:34:36 Process process 2:
Mar 06 03:34:36 Traceback (most recent call last):
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
Mar 06 03:34:36     self.run()
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/multiprocessing/process.py", line 93, in run
Mar 06 03:34:36     self._target(*self._args, **self._kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 243, in _run
Mar 06 03:34:36     getattr(self, test_name)()
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/dist_utils.py", line 71, in new_test_method
Mar 06 03:34:36     rpc.shutdown(graceful=clean_shutdown)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 76, in wrapper
Mar 06 03:34:36     return func(*args, **kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 237, in shutdown
Mar 06 03:34:36     _wait_all_workers()
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 76, in wrapper
Mar 06 03:34:36     return func(*args, **kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 164, in _wait_all_workers
Mar 06 03:34:36     args=(sequence_id, self_worker_name,),
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 76, in wrapper
Mar 06 03:34:36     return func(*args, **kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 526, in rpc_sync
Mar 06 03:34:36     return fut.wait()
Mar 06 03:34:36 RuntimeError: RPC ran for more than 60000 milliseconds and timed out.
Mar 06 03:34:36 Process process 1:
Mar 06 03:34:36 Traceback (most recent call last):
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
Mar 06 03:34:36     self.run()
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/multiprocessing/process.py", line 93, in run
Mar 06 03:34:36     self._target(*self._args, **self._kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 243, in _run
Mar 06 03:34:36     getattr(self, test_name)()
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/dist_utils.py", line 71, in new_test_method
Mar 06 03:34:36     rpc.shutdown(graceful=clean_shutdown)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 76, in wrapper
Mar 06 03:34:36     return func(*args, **kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 237, in shutdown
Mar 06 03:34:36     _wait_all_workers()
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 76, in wrapper
Mar 06 03:34:36     return func(*args, **kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 164, in _wait_all_workers
Mar 06 03:34:36     args=(sequence_id, self_worker_name,),
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 76, in wrapper
Mar 06 03:34:36     return func(*args, **kwargs)
Mar 06 03:34:36   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/api.py", line 526, in rpc_sync
Mar 06 03:34:36     return fut.wait()
Mar 06 03:34:36 RuntimeError: RPC ran for more than 60000 milliseconds and timed out.
Mar 06 03:35:16 Es.
Mar 06 03:35:16 ======================================================================
Mar 06 03:35:16 ERROR [100.090s]: test_torchscript_function_exception (__main__.JitRpcTestWithSpawn)
Mar 06 03:35:16 ----------------------------------------------------------------------
Mar 06 03:35:16 Traceback (most recent call last):
Mar 06 03:35:16   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 175, in wrapper
Mar 06 03:35:16     self._join_processes(fn)
Mar 06 03:35:16   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 256, in _join_processes
Mar 06 03:35:16     self._check_return_codes(elapsed_time)
Mar 06 03:35:16   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 275, in _check_return_codes
Mar 06 03:35:16     raise RuntimeError('Process {} terminated or timed out after {} seconds'.format(i, elapsed_time))
Mar 06 03:35:16 RuntimeError: Process 0 terminated or timed out after 100.07535076141357 seconds
```

cc @ezyang @gchanan @zou3519 @suo @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @xush6528 @jjlilley @osalpekar

**Comment by zhaojuanmao** [comment]
GIL change in #33329 could cause deadlock

**Comment by mrshenli** [comment]
closed by #34505
